crates/apollo-compiler/tests/input_coercion.rs

```diff
@@ -0,0 +1,50 @@
+
+#[test]
```

Comment:
This test won't run as is. It needs to be added to `apollo-compiler/tests/validation/main.rs`. Once added, you will notice rust compiler won't compile this file, and you will need to declare `Schema`, `ExecutableDocument` and `coerce_variable_values`.